### PR TITLE
PLAT-565: test check DT approver != message sender

### PR DESCRIPTION
**Pull request by MikhailZhukov**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: ledger-core/virtual/integration/delegationtoken_test.go [hunk 1]
```diff
@@ -225,3 +225,89 @@ func TestDelegationToken_CheckFailIfWrongApprover(t *testing.T) {
...
+			server.OverrideConveyorFactoryLogContext(inslogger.SetLogger(ctx, logger))
+
+			server.Init(ctx)
+			// increment pulse for VStateReport and VDelegatedCallRequest
+			server.IncrementPulse(ctx)
+
+			jetCoordinatorMock.
+				MeMock.Return(server.GlobalCaller()).
+				QueryRoleMock.Set(
+				func(_ context.Context, _ node.DynamicRole, obj reference.Local, pulse pulse.Number) ([]reference.Global, error) {
```
Comment (ivanshibitov) [inline ledger-core/virtual/integration/delegationtoken_test.go:292]: May be written shorter
QueryRoleMock.Return([]reference.Global{server.GlobalCaller()}, nil)

File: ledger-core/virtual/integration/delegationtoken_test.go [hunk 2]
```diff
@@ -225,3 +225,89 @@ func TestDelegationToken_CheckFailIfWrongApprover(t *testing.T) {
...
+			)
+
+			delegationToken := server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), server.GlobalCaller(), outgoing)
+			reflect.ValueOf(testCase.msg).MethodByName("Reset").Call([]reflect.Value{})
+			insertToken(delegationToken, testCase.msg)
+
+			server.SendPayload(ctx, testCase.msg.(payload.Marshaler))
+			server.WaitIdleConveyor()
+
+			assert.True(t, errorFound)
```
Comment (ivanshibitov) [inline ledger-core/virtual/integration/delegationtoken_test.go:308]: I would use 'require.True'

Comment (MikhailZhukov) [reply ledger-core/virtual/integration/delegationtoken_test.go:308]: This is test for 6 cases. Using access here allow see all failed test cases

File: ledger-core/virtual/integration/delegationtoken_test.go (hunk 1 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/delegationtoken_test.go]: Done

**Pull request merged**